benchmarks/build-native.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+
+# Build a single native benchmark using
+
+set -e
+
+BENCHMARK_DIR="";
+
+for ARG in "$@"; do
+    if [[ -d $ARG ]]; then
+        BENCHMARK_DIR=$ARG;
+        break
+    fi
+done
+
+print_header() {
+    >&2 echo
+    >&2 echo ===== $@ =====
+}
+
+FLAG='--host'
+
+if [[ $BENCHMARK_DIR == "" ]]; then
+    echo "Unknown benchmark directory; usage: ./build-native.sh [--host] <path to benchmark directory>"
+    exit 1
+fi
+
+# Build directly on host
+if [[ $* == *$FLAG* ]]; then
+
+print_header "Build Native Benchmark Using Host"
+(set -x; cd $BENCHMARK_DIR && cargo run --release)
```

Comment:
@abrown It is layered like you said. The bottom layer is all that is needed (the cargo project a) and the other layers are only there for consistency (docker b) or convenience (bash script c). There is really one way to build .. the cargo project. Go into the benchmark directory and type "cargo run". That's it. It is targeting the native host being used so it makes sense to build directly on that native host. But because the wasm is built with Docker and there is the desire for building everything the same way I've added a Docker file which just copies the cargo project, and runs "cargo run" The user would have to copy the binary out of the container. The last layer (the script) is there for convenience and consistency and can run either method depending on whether the --host flag is supplied and it of course copies the binary out of the container when using docker and not host.

Reply:
Will mark this as resolved, but of course can revisit as desired if something different is desired.